### PR TITLE
outdated/install fixes

### PR DESCRIPTION
**Pull request by vladshablinsky**
Fix https://github.com/Homebrew/homebrew/pull/42940 from https://github.com/Homebrew/homebrew/issues/42965

cc @mikemcquaid @xu-cheng

### CODE REVIEW

File: Library/Homebrew/cmd/install.rb [hunk 1]
```diff
@@ -75,14 +75,10 @@ def install
...
-          unless f.linked_keg.symlink?
-            if f.oldname && (HOMEBREW_CELLAR/f.oldname).exist?
-              msg << ", it's just not migrated"
-            elsif !f.keg_only?
-              msg << ", it's just not linked"
-            end
-          end
+          msg << ", it's just not linked" unless f.linked_keg.symlink? || f.keg_only?
           opoo msg
+        elsif f.oldname && (dir = HOMEBREW_CELLAR/f.oldname).exist? && !dir.subdirs.empty?
```
Comment (xu-cheng) [inline Library/Homebrew/cmd/install.rb]: We need to handle the case that oldname is another formula. To do that, I suggest:
* check tap origin 
* allow --force to bypass the migration message.

File: Library/Homebrew/cmd/install.rb [hunk 2]
```diff
@@ -75,14 +75,13 @@ def install
...
-              msg << ", it's just not linked"
-            end
-          end
+          msg << ", it's just not linked" unless f.linked_keg.symlink? || f.keg_only?
           opoo msg
+        elsif f.oldname && (dir = HOMEBREW_CELLAR/f.oldname).exist? && !dir.subdirs.empty? \
+            && f.tap == Tab.for_keg(dir.subdirs.first).tap && !ARGV.force?
+          # Check if the formula we try to install is the same as installed
+          # but not migrated one. If --force passed then install anyway.
+          opoo "#{f.oldname} already installed, it's just not migrated"
```
Comment (xu-cheng) [inline Library/Homebrew/cmd/install.rb]: I think we should mention --force in the message.

Comment (vladshablinsky) [inline Library/Homebrew/cmd/install.rb]: Added
```ruby
puts "You can force install it with `brew install #{f} --force`"
```

**Comment by mikemcquaid** [comment]
:+1: from me. @xu-cheng?

File: Library/Homebrew/cmd/install.rb (hunk 2 above)
Comment (xu-cheng) [inline Library/Homebrew/cmd/install.rb]: Maybe it's better to have:
```ruby
puts "You can migrate formula with `brew migrate #{f}`"
puts "Or you can force install it with `brew install #{f} --force`"
```

**Comment by xu-cheng** [comment]
Need some word tweak, otherwise :+1: from me.

File: Library/Homebrew/cmd/install.rb (hunk 2 above)
Comment (vladshablinsky) [inline Library/Homebrew/cmd/install.rb]: :+1:

**Comment by mikemcquaid** [comment]
Thanks @vladshablinsky!

**Pull request closed without merging**